src/lib/transition_handler/catchup_scheduler.ml

```diff
@@ -30,11 +30,17 @@ module Make (Inputs : Inputs.S) = struct
         , synchronous
         , unit Deferred.t )
         Writer.t
-    ; timeouts:
-        ( (External_transition.Verified.t, State_hash.t) With_hash.t
-        , unit Time.Timeout.t )
-        List.Assoc.t
+        (* `collected_transitins` stores all seen transitions as its keys, and
```

Comment:
Nit: the comments are not properly formatted. The lines after the first line should be indented after the first line. You can use `ocamlformat --wrap-comment` to automatically do this and wrap the comments.